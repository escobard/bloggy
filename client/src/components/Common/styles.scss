@import "../../styles/variables";
@import "../../styles/mixins";

.add-job{
	float: right;
	button{
		background-color: $primary-dark !important;
	}
}