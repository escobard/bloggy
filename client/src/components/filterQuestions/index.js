import React, { Component } from "react"
import { Link } from "react-router-dom"

import { condition } from "./utils"

export default function(questions, selectQuestion, isUnanswered) {
  let answerFilter = questions.filter(question =>
    condition(question, isUnanswered)
  )

  return answerFilter.map((question, index) => {
    let { optionOne, optionTwo, id } = question
    let q1 = optionOne.text
    let q2 = optionTwo.text
    return (
      <Link key={index} to={`/questions/${id}`}>
        <p onClick={() => selectQuestion(question)}>
          {q1} or {q2}
        </p>
      </Link>
    )
  })
}
