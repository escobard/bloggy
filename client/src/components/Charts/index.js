import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import {
  Card,
  CardActions,
  CardHeader,
  CardMedia,
  CardTitle,
  CardText
} from "material-ui/Card";
import FlatButton from "material-ui/FlatButton";
import { Pie } from "react-chartjs-2";

import styles from "./styles.scss"

const data = {
  labels: ["Red", "Green", "Yellow"],
  datasets: [
    {
      data: [300, 50, 100],
      backgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"],
      hoverBackgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"]
    }
  ]
};

export default class Chart extends Component {
  renderCharts = (variant, data) => {
    switch (variant) {
      case "pie":
        return (
          <Fragment>
            <Pie data={data} />
          </Fragment>
        );
      case "polar":
        return;
      case "bar":
        return;
      case "donught":
        return;
      default:
        return <p>No variant!</p>;
    }
  }

  render() {
    let { title, subtitle, description, variant, data } = this.props;
    return (
      <Card className={`chart ${variant}`}>
       { title !== '' ? <CardTitle title={title} subtitle={ subtitle ? subtitle : null} /> : null }
       { description !== '' ? <CardText>{description}</CardText> : null } 
        
        {this.renderCharts(variant, data)}
      </Card>
    );
  }
}

Chart.defaultProps = {
  title: "test title",
  subtitle: "test sub",
  description: "test desc",
  variant: "pie",
  data: {
    labels: ["Red", "Green", "Yellow"],
    datasets: [
      {
        data: [300, 50, 100],
        backgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"],
        hoverBackgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"]
      }
    ]
  }
};

Chart.propTypes = {
  title: PropTypes.string,
  subtitle: PropTypes.string,
  description: PropTypes.string,
  variant: PropTypes.string,
  data: PropTypes.object
};