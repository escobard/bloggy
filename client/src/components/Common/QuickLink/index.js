import React, { Component } from "react";
import PropTypes from "prop-types";
import { RaisedButton, Card } from "material-ui";
import { Link } from "react-router-dom";
import DashIcon from "material-ui/svg-icons/action/dashboard";

import styles from "./styles.scss";

// add images and finish layout of 2x2 for each cta
// <img src="" alt={`${route} icon`} /> --- code for image when icon is replaced
export default class QuickLink extends Component {
	render() {
		let { img, route } = this.props.route;
		return (
			<div className="dashboard-cta col-md-4">
				<Link to={`/dashboard/${route}`}>
					<Card className="card">
						<DashIcon />
						<h4>{route}</h4>
					</Card>
				</Link>
			</div>
		);
	}
}

QuickLink.defaultProps = {
	route: {
		img: "test",
		route: "test"
	}
};

QuickLink.propTypes = {
	routeObject: PropTypes.object
};