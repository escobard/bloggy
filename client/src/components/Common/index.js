import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import FloatingActionButton from "material-ui/FloatingActionButton";
import ContentAdd from "material-ui/svg-icons/content/add";

import styles from "./styles.scss"

export const AddButton = ({ route, open }) => (
	<Link to={route} className="add-job" onClick={open}>
		<FloatingActionButton >
			<ContentAdd />
		</FloatingActionButton>
	</Link>
);

AddButton.defaultProps = {
	route: "/"
};

AddButton.propTypes = {
	route: PropTypes.string
};