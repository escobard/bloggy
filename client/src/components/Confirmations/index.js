import React, { Fragment, Component } from "react";
import PropTypes from "prop-types";

import styles from "./styles.scss";

export class ReviewForm extends Component {
	renderData = reviewData => {
		let reviewKeys = Object.keys(reviewData);
		let reviewValues = Object.values(reviewData);

		return reviewValues.map((data, index) => {
			let label = reviewKeys[index];
			return (
				<div key={index} className="job-field">
					<p className="first">{label}</p>
					<p>{data}</p>
				</div>
			);
		});
	};
	render() {
		console.log(this.props.reviewData);
		return (
			<div className="review-job">
				{this.renderData(this.props.reviewData)}
			</div>
		);
	}
}

ReviewForm.defaultProps = {
	reviewData: "reviewData"
};

ReviewForm.propTypes = {
	reviewData: PropTypes.object
};