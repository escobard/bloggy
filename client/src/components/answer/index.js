import React, { Component } from "react"
import {
    Card,
    CardActions,
    CardTitle,
    CardText,
    RaisedButton
} from "material-ui/Card"

import styles from "./styles.scss"

import { percentage } from "./utils"

export default function({ voteCount, text, percentage, handler }) {
    return (
        <div onClick={handler} className="answer col-md-6 col-sm-12">
            <Card>
                <CardTitle
                    title={percentage}
                    subtitle={`Votes: ${voteCount}`}
                />
                <CardText>{text}</CardText>
            </Card>
        </div>
    )
}
