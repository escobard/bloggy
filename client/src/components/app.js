import React, { Component } from "react";
import { Route } from "react-router-dom";

import Header from "./Header";
import Footer from "./Footer";

import Dashboard from "../containers/dashboard";
import Home from "../containers/home";
import AddQuestion from "../containers/addQuestion";
import Question from "../containers/question"
import Account from "../containers/account"
import Profile from "../containers/profile"
import Survey from "../containers/survey"
import Blog from "../containers/blog"

import styles from "../styles/global.scss"

export default class App extends Component {
    render() {
        return (
            <div>
                <Header />
                <div className="container">
                    <Route exact path="/" component={Home} />
                    <Route path="/dashboard" component={Dashboard} />
                        <Route path="/dashboard/account" component={Account} />
                        <Route path="/dashboard/profile" component={Profile} />
                        <Route path="/dashboard/survey" component={Survey} />
                        <Route path="/dashboard/blog" component={Blog} />
                    <Route exact path="/addquestion" component={AddQuestion} />
                    <Route exact path="/questions/:id" component={Question}/>
                </div>
                <Footer />
            </div>
        );
    }
}