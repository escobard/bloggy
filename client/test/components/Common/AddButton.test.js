import React, { Component } from 'react'
import { AddButton } from '../../../src/components/Common';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'

describe('AddButton snapshot', () => {
	const initialState = {output:100}
	const mockStore = configureStore()
	let store,container

    beforeEach(()=>{
        store = mockStore(initialState)
        container = shallow(<AddButton /> )  
    })

    it('+++ render the AddButton components', () => {
       expect(toJson(container)).toMatchSnapshot()
    });

});