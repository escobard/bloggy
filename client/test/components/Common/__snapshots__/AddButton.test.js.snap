// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton snapshot +++ render the AddButton components 1`] = `
<Link
  className="add-job"
  replace={false}
  to="/"
>
  <FloatingActionButton
    disabled={false}
    mini={false}
    secondary={false}
    zDepth={2}
  >
    <ContentAdd />
  </FloatingActionButton>
</Link>
`;
